packages/service-library/tests/test_archiving_utils.py

```diff
@@ -0,0 +1,237 @@
+        return {k: v for k, v in await asyncio.gather(*tasks)}
+
+
+def full_file_path_from_dir_and_subdirs(dir_path: Path) -> List[Path]:
+    return [x for x in dir_path.rglob("*") if x.is_file()]
+
+
+async def assert_same_directory_content(
+    dir_to_compress: Path, output_dir: Path, inject_relative_path: Path = None
+) -> None:
+    def _relative_path(input_path: Path) -> Path:
+        return Path(str(inject_relative_path / str(input_path))[1:])
+
+    input_set = get_all_files_in_dir(dir_to_compress)
+    output_set = get_all_files_in_dir(output_dir)
+
+    if inject_relative_path is not None:
+        input_set = {_relative_path(x) for x in input_set}
+
+    assert (
+        input_set == output_set
+    ), f"There following files are missing {input_set - output_set}"
+
+    # computing the hashes for dir_to_compress and map in a dict
+    # with the name starting from the root of the directory and md5sum
+    dir_to_compress_hashes = {
+        strip_directory_from_path(k, dir_to_compress): v
+        for k, v in (
+            await compute_hashes(full_file_path_from_dir_and_subdirs(dir_to_compress))
+        ).items()
+    }
+
+    # computing the hashes for output_dir and map in a dict
+    # with the name starting from the root of the directory and md5sum
+    output_dir_hashes = {
+        strip_directory_from_path(k, output_dir): v
+        for k, v in (
+            await compute_hashes(full_file_path_from_dir_and_subdirs(output_dir))
+        ).items()
+    }
+
+    # finally check if hashes are mapped 1 to 1 in order to verify
+    # that the compress/decompress worked correctly
+    for key in dir_to_compress_hashes:
+        assert (
+            dir_to_compress_hashes[key]
+            == output_dir_hashes[_relative_path(key) if inject_relative_path else key]
+        )
+
+
+# end utils
+
+
+@pytest.mark.parametrize(
+    "compress,store_relative_path",
+    [[True, True], [True, False], [False, True], [False, False]],
+)
+async def test_archive_unarchive_same_structure_dir(
+    dir_with_random_content: Path,
+    temp_dir_one: Path,
+    temp_dir_two: Path,
+    compress: bool,
+    store_relative_path: bool,
+):
+    archive_file = temp_dir_one / "archive.zip"
+
+    archive_result = await archive_dir(
+        dir_to_compress=dir_with_random_content,
+        destination=archive_file,
+        store_relative_path=store_relative_path,
+        compress=compress,
+    )
+    assert archive_result is True
+
+    await unarchive_dir(
+        archive_to_extract=archive_file, destination_folder=temp_dir_two
+    )
+
+    await assert_same_directory_content(
+        dir_with_random_content,
+        temp_dir_two,
+        None if store_relative_path else dir_with_random_content,
+    )
+
+
+@pytest.mark.parametrize(
+    "compress,store_relative_path",
+    [[True, True], [True, False], [False, True], [False, False]],
```

Comment:
FYI: check [itertools.product](https://docs.python.org/3/library/itertools.html#itertools.product) to create cartesian products